token/program-2022/src/processor.rs

```diff
@@ -116,24 +116,29 @@ impl Processor {
                 .map_err(|_| Into::<ProgramError>::into(TokenError::InvalidMint))?;
         }
 
-        account.mint = *mint_info.key;
-        account.owner = *owner;
-        account.delegate = COption::None;
-        account.delegated_amount = 0;
-        account.state = AccountState::Initialized;
+        let required_extensions = Self::get_required_account_extensions(mint_info)?;
+        for extension in required_extensions {
+            account.init_extension_from_type(extension)?;
+        }
```

Comment:
This isn't part of our original spec, but what do you think about a check to make sure that there's no padding at the end of the account buffer?

Reply:
Makes a lot of sense.